### PR TITLE
change the plugins catalog resource

### PR DESCRIPTION
**Pull request by barshimi**
Close: #4004 

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by barshimi** [comment]
few notes regard this PR:

- I merged the changes of plugin API service into this branch in order to see full cycle of the process.
- I didn't finished design the css of the dropDown component. we need to refactor the select component to be more flexible.
- I wanted to move usePlugin hook up two levels, so it wont make all call many times. I'll open bug/feature request for that

**Review by morhag90 — APPROVED** [review]

**Comment by yuval-hazaz** [comment]
@barshimi what is the status of this PR? Let's make sure we merge it if it is ready to be merged... It has sat open for two weeks already while we keep drifting away 
let me know if you want to review that together

File: packages/amplication-client/src/Plugins/InstalledPluginSettings.scss [hunk 1]
```diff
@@ -17,6 +17,30 @@
...
+    align-items: flex-start;
+    gap: var(--default-spacing);
+    &:last-child {
+      margin-bottom: 0;
+    }
+  }
+
+  &__label-title {
+    @include regular-12;
+    margin-top: 7px;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Plugins/InstalledPluginSettings.scss:34]: use param

File: packages/amplication-client/src/Plugins/InstalledPluginSettings.scss [hunk 2]
```diff
@@ -17,6 +17,30 @@
...
+  }
+
+  &__menu {
+    width: 500px;
+    .select-menu__modal > div {
+      width: 200px;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Plugins/InstalledPluginSettings.scss:40]: use local params

File: packages/amplication-client/src/Plugins/InstalledPluginSettings.tsx [hunk 3]
```diff
@@ -114,16 +128,54 @@ const InstalledPluginSettings: React.FC<Props> = ({
...
+                        <SelectMenuItem
+                          closeAfterSelectionChange
+                          itemData={pluginVersion}
+                          selected={
+                            pluginVersion.version ===
+                            pluginInstallation.PluginInstallation.version
+                          }
+                          key={pluginVersion.id}
+                          onSelectionChange={(pluginVersion) => {
+                            handlePluginInstalledChange(
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Plugins/InstalledPluginSettings.tsx:158]: version change should not be saved automatically - first show the new settings object and the save button will save all changes including the version selection

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
well done! see comments

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**